api4/plugin.go

```diff
@@ -271,6 +271,11 @@ func getMarketplacePlugins(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if !*c.App.Config().PluginSettings.EnableRemoteMarketplace && !filter.LocalPlugins {
+		c.Err = model.NewAppError("getMarketplacePlugins", "app.plugin.remote_marketplace_disabled.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
```

Comment:
@ali-farooq0 continuing the conversation here, it seems we are just double checking that the web App set these flags properly. It seems error prone to have it handle this extra flag on the client side. Thoughts on just handling the `local_plugins_only` in the server side?

Reply:
@marianunez Yep I like your idea of doing it server side rather than client. `local_plugins_only` will only be used to filter the plugins. `EnableRemoteMarketplace` flag will determine if the server needs to fetch from remote marketplace or not. 👍